x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/util/ShapeUtilTests.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.spatial.util;
+
+import org.apache.lucene.geo.XShapeTestUtil;
+import org.apache.lucene.geo.XYPolygon;
+import org.apache.lucene.geo.XYRectangle;
+import org.elasticsearch.test.ESTestCase;
+
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.lessThan;
+
+public class ShapeUtilTests extends ESTestCase {
+    public void testBox() {
+        XYRectangle geom = XShapeTestUtil.nextBox();
+        assertThat("Geometry minX should be less than maxX", geom.minX, lessThan(geom.maxX));
+        assertThat("Geometry minY should be less than maxY", geom.minY, lessThan(geom.maxY));
+    }
+
+    public void testPolygon() {
+        XYPolygon geom = XShapeTestUtil.nextPolygon();
+        assertThat("Geometry minX should be less than maxX", geom.minX, lessThan(geom.maxX));
+        assertThat("Geometry minY should be less than maxY", geom.minY, lessThan(geom.maxY));
+        assertThat("Geometry area should be non-zero", ShapeTestUtils.area(geom), greaterThan(0.0));
+    }
+
+    public void testFlatRectangle() {
+        XYPolygon geom = new XYPolygon(
```

Comment:
This test exists primarily to demonstrate the bad behaviour of the original failing test. Debugging that was not easy. I think having a failing test is good. In this case since I'm testing the return value of the area function, I do not look for an actual failure, just that the value is below the threshold.

Reply:
that;s good